pkg/delete/delete.go

```diff
@@ -369,3 +372,55 @@ func AllAutomations(ctx context.Context, c automationClient) []error {
 
 	return errs
 }
+
+// AllBuckets collects and deletes objects of type "bucket" using the provided bucketClient.
+//
+// Parameters:
+//   - ctx (context.Context): The context for the operation.
+//   - c (bucketClient): The bucketClient used for listing and deleting objects.
+//
+// Returns:
+//   - []error: A slice of errors encountered during the operation. It may contain listing errors,
+//     deletion errors, or API errors.
+func AllBuckets(ctx context.Context, c bucketClient) []error {
+	var errs []error
+
+	log.WithCtxFields(ctx).WithFields(field.Type("bucket")).Info("Collecting objects of type %q...", "bucket")
+	response, err := c.List(ctx)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	if err, ok := response.AsAPIError(); ok {
+		errs = append(errs, err)
+	}
+
+	log.WithCtxFields(ctx).WithFields(field.Type("bucket")).Info("Deleting %d objects of type %q...", len(response.Objects), "bucket")
+	for _, obj := range response.Objects {
+		var bucketName struct {
+			BucketName string `json:"bucketName"`
+		}
+
+		// exclude builtin bucket names, they cannot be deleted anyway
+		if strings.HasPrefix(bucketName.BucketName, "dt_") || strings.HasPrefix(bucketName.BucketName, "default_") {
```

Comment:
that needs to happen after unmarshalling, doesn't it? 😅 `bucketName.BucketName` will still be zero initialized here